src/runtime_src/core/tools/xbmgmt2/xbmgmt

```diff
@@ -1,29 +1,84 @@
 #!/bin/bash
 
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2020-2021 Xilinx, Inc.
+#
+
 # -- Detect a Windows environment and automatically switch to the .bat file
 if [[ "`uname`" == windows32* ]] || [[ "`uname`" == CYGWIN* ]] ; then
   trap "" INT
   "$0.bat" "$@"
   exit $?
 fi
 
+DEFAULT=xbmgmt
+XBMGMT=xbmgmt
+XBMGMT2=xbmgmt2
+
+# Table to resolve reset sub-command based on switch passed to reset
+declare -A RSTOPT=( \
+        ["--hot"]="xbmgmt" \
+        ["--kernel"]="xbmgmt" \
+        ["--ecc"]="xbmgmt" \
+        ["--softkernel"]="xbmgmt" \
+        ["--ert"]="xbmgmt" \
+        ["--card"]="xbmgmt" \
+        ["--force"]="xbmgmt" \
+        ["--device"]="xbmgmt2" \
+        ["--type"]="xbmgmt2" \
+        ["--help"]="xbmgmt2" \
+        ["-d"]="xbmgmt2" \
+        ["-r"]="xbmgmt2" \
+        ["-h"]="xbmgmt2" \
+    )
+
+# In the following table "reset" points to a second table by name
+# Note that reset is valid both for classic and next generation tool
+# and hence requires a second lookup to resolve between xbmgmt and xbmgmt2
+declare -A COMMANDS=( \
+        ["help"]="XBMGMT" \
```

Comment:
shouldn't this default to xbmgmt2 as well? Maybe we should come up with a better way to differentiate between xbmgmt v1 and v2 help. It might be confusing as to why "--" prints a different help

Reply:
The default is still xbmgmt and I was hoping that the current users and scripts do not see any change. So `xbgmt help` should continue to work as before.
`DEFAULT=xbmgmt`

Reply:
Ok. That sounds good.